quinn-udp/src/unix.rs

```diff
@@ -116,22 +97,28 @@ fn init(io: &std::net::UdpSocket) -> io::Result<()> {
 
     let addr = io.local_addr()?;
 
-    // macos and ios do not support IP_RECVTOS on dual-stack sockets :(
-    if addr.is_ipv4() || ((!cfg!(any(target_os = "macos", target_os = "ios"))) && !only_v6(io)?) {
-        let on: libc::c_int = 1;
-        let rc = unsafe {
-            libc::setsockopt(
-                io.as_raw_fd(),
-                libc::IPPROTO_IP,
-                libc::IP_RECVTOS,
-                &on as *const _ as _,
-                mem::size_of_val(&on) as _,
-            )
-        };
-        if rc == -1 {
-            return Err(io::Error::last_os_error());
+    // mac and ios do not support IP_RECVTOS on dual-stack sockets :(
+    // older macos versions also don't have the flag and will error out if we don't ignore it
+    let on: libc::c_int = 1;
+    let rc = unsafe {
+        libc::setsockopt(
+            io.as_raw_fd(),
+            libc::IPPROTO_IP,
+            libc::IP_RECVTOS,
+            &on as *const _ as _,
+            mem::size_of_val(&on) as _,
+        )
+    };
+    if rc == -1 {
+        let err = io::Error::last_os_error();
+        // ignore on macos/ios but keep linux behavior the same as before
+        if cfg!(any(target_os = "macos", target_os = "ios")) {
+            tracing::warn!("Ignoring error setting IP_RECVTOS on socket: {:?}", err);
```

Comment:
Would it make more sense for this to be `info` or even `debug`? By design, Quinn is well-behaved if ECN data is missing, and it's routinely missing due to the data being stripped along the network path even if both hosts support it.

Reply:
yeah wasn't sure what was best, maybe debug if it's not important enough?